src/plugins/intel_gpu/include/intel_gpu/primitives/gemm.hpp

```diff
@@ -38,42 +38,114 @@ struct gemm : public primitive_base<gemm> {
+         const bool transpose_input0,
+         const bool transpose_input1,
          const float alpha = 1.0f,
          const float beta = 0.0f,
          const size_t input_rank = 4,
          const size_t weight_rank = 4,
          const padding& output_padding = padding())
         : primitive_base(id, inputs, {output_padding}, {optional_data_type{ data_type }}),
-          transpose_input0(transpose_input0),
-          transpose_input1(transpose_input1),
+          transpose_input0(transpose_input0 ? 1 : 0),
+          transpose_input1(transpose_input1 ? 1 : 0),
           alpha(alpha),
           beta(beta),
           input_rank(input_rank),
           weight_rank(weight_rank) {
         if (inputs.size() != 2 && inputs.size() != 3) {
             throw std::invalid_argument("Invalid inputs count - gemm expects either two or three inputs");
         }
+
+        auto get_tranaposed_order = [] (size_t rank, bool transposed) {
+            std::vector<int64_t> order(rank);
+            std::iota(order.begin(), order.end(), 0);
+            if (transposed)
+                std::swap(order[rank - 1], order[rank - 2]);
+            return order;
+        };
+
+        input0_order = get_tranaposed_order(input_rank, transpose_input0);
+        input1_order = get_tranaposed_order(weight_rank, transpose_input1);
+        output_order = {};
+    }
+
+    /// @brief Constructs gemm layer.
+    /// @brief Primitive id containing first matrix
+    /// @brief Primitive id containing second matrix
+    /// @brief Transposed order of first input matrix
+    /// @brief Transposed order of second input matrix
+    /// @brief Transposed order of output matrix
+    /// @brief Variable containing ALPHA parameter
+    /// @brief Variable containing BETA parameter
+    gemm(const primitive_id& id,
+         const std::vector<input_info>& inputs,
+         const data_types data_type,
+         const std::vector<int64_t>& input0_order = {0, 1, 2, 3},
+         const std::vector<int64_t>& input1_order = {0, 1, 2, 3},
+         const std::vector<int64_t>& output_order = {},
+         const float alpha = 1.0f,
+         const float beta = 0.0f,
+         const padding& output_padding = padding())
+        : primitive_base(id, inputs, {output_padding}, {optional_data_type{ data_type }}),
+          input0_order(input0_order),
+          input1_order(input1_order),
+          output_order(output_order),
+          alpha(alpha),
+          beta(beta),
+          input_rank(input0_order.size()),
+          weight_rank(input1_order.size()) {
+        if (inputs.size() != 2 && inputs.size() != 3) {
+            throw std::invalid_argument("Invalid inputs count - gemm expects either two or three inputs");
+        }
+
+        auto get_transpose_mode = [](const std::vector<int64_t>& order_idx) {
+            int64_t rank = order_idx.size() - 1;
+
+            if (rank == order_idx[rank]) {
+                // normal
+                return 0;
```

Comment:
Could you set these as enumeration for better readability?

Reply:
And can we merge the two constructors to one? (e.g., the below consturctor can cover..?)

Reply:
I added an enumeration to represent the transposition type as you reviewed. Thank you.